src/arrays.js

```diff
@@ -1,61 +1,93 @@
 const getNthElement = (index, array) => {
-  // your code here
+  if (index > array.length - 1) {
+    index -= array.length;
+  }
+  return array[index];
 };
 
-const arrayToCSVString = array => {
-  // your code here
-};
+const arrayToCSVString = array => array.toString();
 
-const csvStringToArray = string => {
-  // your code here
-};
+const csvStringToArray = string => string.split(',');
 
 const addToArray = (element, array) => {
-  // your code here
+  array.push(element);
 };
 
-const addToArray2 = (element, array) => {
-  // your code here
-};
+const addToArray2 = (element, array) => array.concat(element);
 
-const removeNthElement = (index, array) => {
-  // your code here
-};
+const removeNthElement = (index, array) => array.splice(index, 1);
 
 const numbersToStrings = numbers => {
-  // your code here
+  const arr = numbers.map(item => {
+    return item.toString();
+  });
+  return arr;
 };
 
 const uppercaseWordsInArray = strings => {
-  // your code here
+  const arr = strings.map(item => {
+    return item.toUpperCase();
+  });
+  return arr;
 };
 
 const reverseWordsInArray = strings => {
-  // your code here
+  const arr = strings.map(item => {
+    return item
+      .split('')
+      .reverse()
+      .join('');
+  });
+  return arr;
 };
 
 const onlyEven = numbers => {
-  // your code here
+  const arr = numbers.filter(item => {
+    return item % 2 === 0;
+  });
+  return arr;
 };
 
 const removeNthElement2 = (index, array) => {
-  // your code here
+  const arr = array.filter((_, elementIndex) => elementIndex !== index);
+  return arr;
 };
 
 const elementsStartingWithAVowel = strings => {
-  // your code here
+  const arr = strings.filter(items => {
+    return (
+      items[0].includes('a') ||
+      items[0].includes('A') ||
+      items[0].includes('e') ||
+      items[0].includes('E') ||
+      items[0].includes('i') ||
+      items[0].includes('I') ||
+      items[0].includes('o') ||
+      items[0].includes('O') ||
+      items[0].includes('u') ||
+      items[0].includes('U')
+    );
+  });
+  return arr;
```

Comment:
This is a pretty verbose way of doing this! Filtering is the right approach, but theres an awful lot of duplication here that doesnt need to be. The cleanest way by far to do this is with a regular expression, which looks like this: /[aeiou]/i
This is a language agnostic way to do pattern recognition, which is ultimately what you are trying to do here. Basically the contents of the square brackets are the characters you want to match, and we can return only the elements we want by using the built in test function like this:
`const result = /[aeiou]/i.test(yourStringHere)`
if `yourStringHere` contains any of the values in the square brackets, it will pass. Since we need to know if just the _first_ letter is a vowel, we can add the `^` symbol before the pattern. This just means 'match this from the start of the string'.

So when combining it with filter as in the above method, you get this:
```
  return strings.filter(string => /^[aeiou]/i.test(string));
```
There are other ways to do this such as looping over the strings and then looping again over an array of vowels to find matches like that, but its a long way round and inefficient. Im sure you'll be seeing regex again at some point in soon, dont worry if its a bit confusing! Its one of those things that gets easier with time but is incredibly powerful once you get the hang of it